File: code/java/剑指offer58-1.java
class Solution {
    public String reverseWords(String s) {
        String[] temp = s.split(" ");
        StringBuffer sb = new StringBuffer();
        for(int i=temp.length-1;i>=0;i--){
            if(temp[i].strip()!=""){
                sb.append(temp[i].strip());
                sb.append(" ");
            }
        }
        if(sb.length()>0) sb.deleteCharAt(sb.length()-1);
        return sb.toString();
    }
}

File: code/python/剑指offer58-1.py
class Solution:
    def reverseWords(self, s: str) -> str:
        temp = s.split(' ')
        temp = [t.strip() if t.strip()!='' else "" for t in temp]
        temp_ = []
        for t in temp:
            if t!="":
                temp_.append(t)
        return " ".join(temp_[::-1])